### ISSUE TITLE
Scaler.fit() Assumes Model Outputs Are Probabilities

### ISSUE THREAD

**Issue by Arthur-Chiron** [ENSTA-U2IS-AI/torch-uncertainty#179]
In the `Scaler` class, the `fit()` method currently assumes that the model's outputs are probabilities. This assumption is evident in the following line (line 89):
```python
all_logits = all_logits.clamp(self.eps, 1 - self.eps)
```
This operation is only valid if `all_logits` actually contains probabilities, not raw logits. However, in many cases, users design their models to output raw logits and apply a softmax or sigmoid function separately during evaluation or loss computation.

*Suggestion*

Enhance the `fit()` method to support models that return either logits or probabilities. One possible approach is to:
- Automatically detect whether the output appears to be logits or probabilities.
- Or, introduce a parameter (e.g. `input_is_logits: bool`) to let the user explicitly specify the type of output.

This change would improve compatibility and make the `Scaler` class more flexible for a wider range of use cases.

Let me know if you'd like help implementing this change too.

**Comment by giannifranchi** [comment]
Thank you very much for your comment — we really appreciate your feedback. We’ll take the time to analyze it carefully.

To be honest, we’ve observed some unexpected behavior with temperature scaling, with the ECE and ACE  that do not show any improvement. If you have any insights or suggestions on why this might be happening, we’d be very interested to hear them.

**Comment by o-laurent** [comment]
Hello @Arthur-Chiron, 

Thank you for your feedback!

There is indeed a bug in the current implementation, which will be fixed in a few minutes with version 0.5.1. As of 0.5.1, the scalers will work again with logits in the multiclass setting and probabilities in the binary case. However, you are right that we could make this behavior at least more transparent or more general, as you suggest.

**Comment by o-laurent** [comment]
The new version should be available on PyPI. The updated tutorials will be online in 15 minutes. Let's keep the issue open to work on this. @Arthur-Chiron, if you have time, I'll gladly work with you on a dedicated PR, but I can also do it on my end. Up to you! At least now the code works as intended, despite the limitations that you highlighted.

**Comment by Arthur-Chiron** [comment]
Hello @giannifranchi, @o-laurent,

Thanks for your quick feedback!

I realized that my initial issue was theoretically incomplete. To the best of my knowledge, all calibration methods — whether binary (e.g. **Platt Scaling**) or multiclass (e.g. **Temperature Scaling**, **Vector Scaling**, **Matrix Scaling**) — operate on the **logits** $z_i$, not on the probabilities $p_i$. These methods rescale the logits **before** applying a softmax, sigmoid, or any other logit-to-probability transformation.

Because of this, I believe the implementation should **never** take probabilities as input.

That said, many practitioners include a `Softmax` or `Sigmoid` layer at the end of their models (especially when using `torch.nn.Sequential`). In those cases, the output of:

```python
logits = self.model(inputs.to(self.device))
```

would actually be **probabilities**, not logits.

To address this cleanly, I think the best solution is to require `fit()` to take logits and labels directly, and let the user extract the logits themselves:

```python
temp_scaler = TemperatureScaler()
all_logits = []
all_labels = []
with torch.no_grad():
    for inputs, labels in calib_loader:
        logits = model(inputs)  # Assumes the model returns logits (or use a hook to get them)
        all_logits.append(logits)
        all_labels.append(labels)
    all_logits = torch.cat(all_logits)
    all_labels = torch.cat(all_labels)

temp_scaler.fit(all_logits, all_labels)
temperature = temp_scaler.temperature[0].item()
```

I realize this would significantly change how the `Scaler` class works, but I believe it’s the most robust approach.

An alternative would be to keep the current behavior and simply add a warning: if the model outputs probabilities instead of logits, calibration results **will** be invalid.

Let me know what you think — I’d be happy to collaborate on a PR! (It would actually be my first one, so I’m still learning how the process works 🙂)
